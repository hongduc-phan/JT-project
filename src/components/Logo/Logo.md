Normal (Get the first alphabet of the company)

```js
const Logo = require('./').default;

<Logo company="Expresso Media" />;
```

With Logo

```js
const Logo = require('./').default;

<Logo
  company="JuzTalent"
  logo="https://juztalent.com/wp-content/uploads/2017/01/cropped-juztalent.png"
/>;
```
