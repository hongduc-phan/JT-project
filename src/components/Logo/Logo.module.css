.logo {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 2.25rem;
  height: 2.25rem;
  border-radius: 50%;
}

.logoWithIcon {
  width: unset;
  border-radius: 0;
}

.logoDefault {
  font-weight: var(--font-weight-semibold);
  line-height: 1.33;
  background-color: #84cdf0;
}
