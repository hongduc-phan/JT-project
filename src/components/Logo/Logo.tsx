import React, {AllHTMLAttributes} from 'react';
import clsx from 'clsx';

import Typo, {TypoVariants, TypoColors, TypoAlignment} from '../Typo';

import styles from './Logo.module.css';

export interface ILogoProps extends AllHTMLAttributes<HTMLDivElement> {
  logo?: string;
  company: string;
  className?: string;
}

function getFirstCharacter(text: string) {
  if (!text) {
    return '';
  }
  return text[0].toLocaleUpperCase();
}

const Logo = ({logo, company, className, ...others}: ILogoProps) => {
  return (
    <div className={className} {...others}>
      {logo ? (
        <img
          className={clsx(styles.logo, styles.logoWithIcon)}
          src={logo}
          alt={company}
        />
      ) : (
        <Typo
          variant={TypoVariants.h4}
          color={TypoColors.white}
          align={TypoAlignment.middle}
          className={clsx(styles.logo, styles.logoDefault)}
        >
          {getFirstCharacter(company)}
        </Typo>
      )}
    </div>
  );
};

export default Logo;
