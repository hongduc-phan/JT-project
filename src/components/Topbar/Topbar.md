```js
const {Provider} = require('react-redux');
const Topbar = require('./').default;
const Logo = require('../Logo').default;
const Typo = require('../Typo').default;
const store = require('../../store').default;
const {TypoVariants, TypoColors, TypoAlignment} = require('../Typo');
const UserNavigationContainer = require('../../containers/UserNavigationContainer')
  .default;

<Provider store={store}>
  <Topbar>
    <Logo
      company="JuzTalent"
      logo="https://juztalent.com/wp-content/uploads/2017/01/cropped-juztalent.png"
    />
    <Typo
      tag="div"
      variant={TypoVariants.h4}
      color={TypoColors.greyDark}
      align={TypoAlignment.middle}
      className="topbar__title"
    >
      JusTalent
    </Typo>
    <UserNavigationContainer />
  </Topbar>
</Provider>;
```
