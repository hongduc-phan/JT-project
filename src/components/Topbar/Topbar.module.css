.topbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 1.75rem;
  padding: 0.875rem 1.5rem;
  background-color: var(--white);
  box-shadow: 0 1px 3px 0 rgba(63, 63, 68, 0.15),
    0 0 0 1px rgba(63, 63, 68, 0.05);
}
