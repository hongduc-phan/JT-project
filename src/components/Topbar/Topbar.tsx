import React, {ReactNode, AllHTMLAttributes} from 'react';
import clsx from 'clsx';

import styles from './Topbar.module.css';

export interface ITopbarProps extends AllHTMLAttributes<HTMLDivElement> {
  children: ReactNode;
  className?: string;
}

const Topbar = ({children, className, ...others}: ITopbarProps) => {
  return (
    <div className={clsx(styles.topbar, className)} {...others}>
      {children}
    </div>
  );
};

export default Topbar;
