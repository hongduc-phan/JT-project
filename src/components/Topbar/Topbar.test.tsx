import React from 'react';
import renderer from 'react-test-renderer';

import Topbar from './Topbar';
import Logo from '../Logo';

describe('<Topbar />', () => {
  let topBar: any;

  beforeEach(() => {
    topBar = (
      <Topbar>
        <Logo company="Juztalent" />
      </Topbar>
    );
  });

  it('Should match topbar snapshot', () => {
    const tree = renderer.create(topBar).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
