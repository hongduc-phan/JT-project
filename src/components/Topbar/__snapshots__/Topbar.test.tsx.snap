// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Topbar /> Should match topbar snapshot 1`] = `
<div
  className="topbar"
>
  <div>
    <span
      className="typo logo logoDefault h4 white middle"
    >
      J
    </span>
  </div>
</div>
`;
